tutorials/plugins/editor/inspector_plugins.rst

```diff
@@ -8,17 +8,48 @@ editing properties. This tutorial explains how to use the
 :ref:`class_EditorInspectorPlugin` and :ref:`class_EditorProperty` classes to
 write such plugins with the example of creating a custom value editor.
 
-.. note::
+Setup
+-----
+
+Just like :ref:`doc_making_plugins`, we start out by making a new plugin,
+getting a ``plugin.cfg`` file created, and start with our
+:ref:`class_EditorPlugin`.  However, instead of using
+``add_custom_node`` or ``add_control_to_dock`` we'll use
+``add_inspector_plugin``.
+
+
+.. tabs::
+  .. code-tab:: gdscript GDScript
+    # MyEditorPlugin.gd
+
+    tool extends EditorPlugin
+
+    var plugin: EditorInspectorPlugin
+
+    func _enter_tree() -> void:
+        # EditorInspectorPlugin is a resource, so we use `new()` instead of `instance()`
+        plugin = preload("res://addons/MyPlugin/MyInspectorPlugin.gd").new()
+        add_inspector_plugin(plugin)
+
+    func _exit_tree() -> void:
+        remove_inspector_plugin(plugin)
+
+
 
-    To register these scripts as a new editor plugin, you have to create a
-    ``plugin.cfg`` file as described in :ref:`doc_making_plugins`.
 
 EditorInspectorPlugin
 ---------------------
 
-We start by creating a script extending the :ref:`class_EditorInspectorPlugin`
-class. This is needed to initialize the plugin and add the custom property
-editor that we'll later define.
+To actually connect into the Inspector, we create a
+:ref:`class_EditorInspectorPlugin` class.  This script provides
+the "hooks" into the inspector.  It is used to check if we should enable
```

Comment:
What do you mean by "hooks" I don't understand can you expand.

Reply:
Hmm... callback?  The way that this works is you set it up, and the editor will call the functions within the EditorInspectorPlugin while it goes through the process of building the UI for the inspector.

Generally, I've always referred to this as things that "hook into" an existing flow to modify it.   I'm by no means married to that though, so if we think of a better way of explaining it, then I'd be super happy!  This is probably one of the harder systems in here, compared to most other UI work!

Reply:
I think this explanation now makes more sense

So maybe something like 

To actually connect into the Inspector, we create a
:ref:`class_EditorInspectorPlugin` class.  This script the "hooks into" the inspector.  
Thanks to that the editor will call the functions within the EditorInspectorPlugin 
while it goes through the process of building the UI for the inspector. 
It is used to check if we should enable ourselves for any :ref:`class_Object` 
that is currently in the inspector (including any :ref:`class_Resource` that is embedded!).

Or something similar just to explain what you understand by providing hooks